scripts/convert-idl.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Transformer = require('webidl2js');
+
+const allProperties = require('../lib/allProperties.js');
+const allExtraProperties = require('../lib/allExtraProperties.js');
+const parsers = require('../lib/parsers.js');
+
+const srcDir = path.resolve(__dirname, '../src');
+const implDir = path.resolve(__dirname, '../lib');
+const outputDir = implDir;
+
+const propertyNames = Array.from(allProperties)
+  .concat(
+    Array.from(allExtraProperties).filter(prop => prop !== 'css-float' && !allProperties.has(prop))
+  )
+  .sort();
+
+const camelCasedAttributes = propertyNames.map(n => parsers.dashedToCamelCase(n));
+const webkitCasedAttributes = propertyNames
+  .filter(n => n.startsWith('-webkit-'))
+  .map(n => parsers.dashedToCamelCase(n.substring(1)));
+const dashedAttributes = propertyNames.filter(n => n.includes('-'));
+
+{
+  // TODO: This should be natively supported by WebIDL2JS's Transformer
+  // https://github.com/jsdom/webidl2js/issues/188
+  const genIDL = fs.createWriteStream(
+    path.resolve(__dirname, '../src/CSSStyleDeclaration-properties.webidl'),
+    {
+      encoding: 'utf-8',
+    }
+  );
+
+  const dateToday = new Date();
+  genIDL.write(
+    `// autogenerated - ${dateToday.getMonth() +
+      1}/${dateToday.getDate()}/${dateToday.getFullYear()}
```

Comment:
Let's make it scarier.

```suggestion
  genIDL.write(
    `// autogenerated by scripts/convert-idl.js. do not edit! ${new Date().toISOString()}
```